src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js

```diff
@@ -1,16 +1,107 @@
 // @flow
 import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+import noop from 'lodash/noop';
+
+import LoadingIndicator from '../../../../components/loading-indicator';
+import PlainButton from '../../../../components/plain-button';
+import ActivityThreadReplies from './ActivityThreadReplies';
+
+import type { GetAvatarUrlCallback, GetProfileUrlCallback } from '../../../common/flowTypes';
+import type { Translations } from '../../flowTypes';
+import type { SelectorItems, User } from '../../../../common/types/core';
+import type { Comment as CommentType } from '../../../../common/types/feed';
+
+import messages from './messages';
+import './ActivityThread.scss';
 
 type Props = {
     children: React.Node,
+    currentUser?: User,
+    getAvatarUrl: GetAvatarUrlCallback,
+    getMentionWithQuery?: Function,
+    getUserProfileUrl?: GetProfileUrlCallback,
     hasReplies: boolean,
+    id: string,
+    mentionSelectorContacts?: SelectorItems<>,
+    onGetReplies?: (id: string) => void,
+    onReplyDelete?: Function,
+    onReplyEdit?: Function,
+    replies?: Array<CommentType>,
+    repliesLoading?: boolean,
+    total_reply_count?: number,
+    translations?: Translations,
 };
 
-const ActivityThread = ({ children, hasReplies }: Props) => {
-    if (hasReplies) {
-        return <div className="bcs-ActivityThread">{children}</div>;
+const ActivityThread = ({
+    children,
+    currentUser,
+    getAvatarUrl,
+    getMentionWithQuery,
+    getUserProfileUrl,
+    hasReplies,
+    id,
+    mentionSelectorContacts,
+    onGetReplies = noop,
+    onReplyDelete = noop,
+    onReplyEdit = noop,
+    replies = [],
+    repliesLoading,
+    total_reply_count = 0,
+    translations,
+}: Props) => {
+    const [isExpanded, setIsExpanded] = React.useState(false);
+
+    const toggleButtonLabel = isExpanded ? messages.hideReplies : messages.getMoreReplies;
+    const repliesToLoadCount = total_reply_count - 1;
```

Comment:
Can this lead to a negative number? Should we use `Math.max(total_reply_count - 1, 0)`?